### DISCUSSION TITLE
How much space does 'zopen tools'

### DISCUSSION THREAD

**Discussion by MikeFultonDev** [zopencommunity/meta#446] [Reference]
# How much space does ‘zopen’ need?

TL;DR:

To install zopen with only the tools necessary to use zopen itself: ~125MB
To install basic tools (git, bash, curl, make): ~450MB
To install the tools necessary to build git from source: ~875MB

------------------------------------------------------------------------------------------------------

## Details:
To install zopen with only the tools necessary to use zopen itself: 

## Base:

**zopen list --installed**
```
Package                                Installed                              File                                   Releaseline
curl                                   8.1.2                                  curl-8.1.2.20230912_215945.zos         n/a
jq                                     1.6                                    jq-1.6.20230720_160648.zos             STABLE
meta                                   0.3.0-dt-stable                        meta-dt                                n/a

du -k zopen-test
…
```
**126636** zopen-test

## 8 core tools:

**zopen install -y git bash curl make less ncurses perl m4**

**zopen list --installed**

```
Package                                Installed                              File                                   Releaseline
bash                                   5.2.15                                 bash-5.2.15.20230920_202747.zos        n/a
curl                                   8.1.2                                  curl-8.1.2.20230912_215945.zos         n/a
git                                    2.41.0                                 git-2.41.0.20230923_021248.zos         n/a
jq                                     1.6                                    jq-1.6.20230720_160648.zos             STABLE
less                                   643                                    less-643.20230909_075924.zos           n/a
m4                                     1.4.19                                 m4-1.4.19.20230920_223415.zos          n/a
make                                   4.4.1                                  make-4.4.1.20230921_020252.zos         n/a
meta                                   0.3.0-dt-stable                        meta-dt                                n/a
ncurses                                6.3                                    ncurses-6.3.20230914_231534.zos        n/a
perl                                   blead.20230605_161457                  perl5-blead.20230605_161457.zos        STABLE

du -k zopen-test
…
```

**447679** zopen-test

## Tools to build 'git'

**git clone [git@github.com:ZOSOpenTools/gitport.git](mailto:git@github.com:ZOSOpenTools/gitport.git)** 

**cd gitport; zopen build -sc -s # do not actually do the build, but install tools necessary to build ‘git’**

**zopen list --installed**

```
Package                                Installed                              File                                   Releaseline
autoconf                               2.71.20230604_225130                   autoconf-2.71.20230604_225130.zos      n/a
automake                               1.16.5                                 automake-1.16.5.20230913_013426.zos    n/a
bash                                   5.2.15                                 bash-5.2.15.20230920_202747.zos        n/a
comp_xlclang                           2.4.1                                  comp_xlclang.20230921_001414.zos       n/a
coreutils                              9.1                                    coreutils-9.1.20230922_223044.zos      n/a
curl                                   8.1.2                                  curl-8.1.2.20230912_215945.zos         n/a
diffutils                              3.8                                    diffutils-3.8.20230921_213028.zos      n/a
expat                                  2.4.9.20230922_212820                  expat-2.4.9.20230922_212820.zos        n/a
gettext                                0.21                                   gettext-0.21.20230922_120117.zos       n/a
git                                    2.41.0                                 git-2.41.0.20230923_021248.zos         n/a
gzip                                   1.12                                   gzip-1.12.20230920_221035.zos          n/a
help2man                               1.49.2                                 help2man-1.49.2.20230922_184540.zos    n/a
jq                                     1.6                                    jq-1.6.20230720_160648.zos             STABLE
less                                   643                                    less-643.20230909_075924.zos           n/a
libpcre2                               10.42                                  pcre2-10.42.20230721_001051.zos        n/a
m4                                     1.4.19                                 m4-1.4.19.20230920_223415.zos          n/a
make                                   4.4.1                                  make-4.4.1.20230921_020252.zos         n/a
meta                                   0.3.0-dt-stable                        meta-dt                                n/a
ncurses                                6.3                                    ncurses-6.3.20230914_231534.zos        n/a
openssl                                3.1.2.20230828_170300                  openssl-heads.openssl-3.1.2.20230828_170300.zos n/a
perl                                   blead.20230605_161457                  perl5-blead.20230605_161457.zos        STABLE
sed                                    4.8                                    sed-4.8.20230915_030126.zos            n/a
tar                                    1.34                                   tar-1.34.20230922_124947.zos           n/a
texinfo                                6.8                                    texinfo-6.8.20230923_031801.zos        n/a
xz                                     5.2.5                                  xz-5.2.5.20230921_020418.zos           n/a
zlib                                   1.2.13                                 zlib-heads.v1.2.13.20230921_020803.zos n/a
zoslib                                 2.1.0                                  zoslib-zopen.20230922_150106.zos       n/a

du -k zopen-test
…
```

**874027** zopen-test

**Comment by MikeFultonDev** [comment]
This number will change over time... It is current as of the latest tools on September 24th, 2023
